### PR TITLE
Feat: port other packets

### PR DESCRIPTION
**Pull request by cire3wastaken**

### CODE REVIEW

File: src/main/java/prelude/protocol/S2CPacket.java [hunk 1]
```diff
@@ -1,11 +1,20 @@
 package prelude.protocol;
 
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+
 public abstract class S2CPacket extends Packet {
+    public static final Map<Byte, Class<? extends S2CPacket>> packets = new HashMap<>();
```
Comment (ProdPreva1l) [inline src/main/java/prelude/protocol/S2CPacket.java:12]: I don't think a map needs to be type generic, the abstract type should work fine here

File: src/main/java/prelude/protocol/S2CPacket.java [hunk 2]
```diff
@@ -1,11 +1,20 @@
...
+
     public static final byte SERVER_HANDSHAKE_ID = 0;
     public static final byte MOD_STATUS_ID = 1;
     public static final byte RESPAWN_ANCHOR_UPDATE_ID = 2;
     public static final byte SERVER_TPS_ID = 3;
```
Comment (ProdPreva1l) [inline src/main/java/prelude/protocol/S2CPacket.java:18]: Perhaps a separate enum for packet ids?

**Review by ProdPreva1l — CHANGES_REQUESTED** [review]
I like the tests, could be more robust but they work for now

**Comment by cire3wastaken** [comment]
@ProdPreva1l I've decided to use something with probably invoking Class#getDeclaredConstructor#newInstance via grabbing the Class type through a Map<Byte, Class>, and having packets implement a loadData method, which basically loads the data from an input stream into itself

It will look something like this, pretty similar to what Minecraft uses internally and what I planned to use in the original Protocol subproject of Prelude. I've checked, TeaVM supports using Class#newInstance
```java
Map<Byte, Class<? extends S2CPacket> map = ...;

static void handlePacket(byte[] bytes) {
    Class<? extends S2CPacket> clazz = map.get(bytes[0]);
    S2CPacket pkt = clazz.newInstance();
    pkt.loadData(new ByteArrayInputStream(bytes));
    pkt.processSelf(packetProcesserC2S);
}
```

**Comment by ProdPreva1l** [comment]
@cire3wastaken i think it might be easier to have a map of Byte to packet type enum and then a switch for the enum to a builder of the packet

File: src/main/java/prelude/protocol/S2CPacket.java (hunk 1 above)
Comment (cire3wastaken) [reply src/main/java/prelude/protocol/S2CPacket.java:12]: ok

**Comment by cire3wastaken** [comment]
> @cire3wastaken i think it might be easier to have a map of Byte to packet type enum and then a switch for the enum to a builder of the packet

I'm using the same method minecraft uses, using Class#newInstance

File: src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            String modId = StreamUtils.readASCII(is.read(), is);
+            ModStatus modStatus = ModStatus.from((byte) is.read());
+
+            if (modStatus == null)
+                throw new InvalidPacketException("Constructed MOD_STATUS_PACKET has an illegal ModStatus!");
+
+            this.modIdentifier = modId;
+            this.modStatus = modStatus;
+        } catch (Exception e) {
+            if (e instanceof InvalidPacketException)
```
Comment (ProdPreva1l) [inline src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java]: Just do stagnant catches

**Review by ProdPreva1l — CHANGES_REQUESTED** [review]

File: src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java (hunk 3 above)
Comment (cire3wastaken) [reply src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java]: smh

Comment (cire3wastaken) [reply src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java]: nah im too lazy

Comment (cire3wastaken) [reply src/main/java/prelude/protocol/packets/s2c/ModStatusPacket.java]: 💀 fine

**Pull request merged**